spring-cloud-netflix-eureka-client/src/test/java/org/springframework/cloud/netflix/eureka/http/WebClientTransportClientFactoryTest.java

```diff
@@ -16,23 +16,57 @@
 
 package org.springframework.cloud.netflix.eureka.http;
 
+import java.time.Duration;
+
 import com.netflix.discovery.shared.resolver.DefaultEndpoint;
 import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+import reactor.core.publisher.Mono;
 
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.reactive.function.client.ClientRequest;
+import org.springframework.web.reactive.function.client.ClientResponse;
+import org.springframework.web.reactive.function.client.ExchangeFunction;
 import org.springframework.web.reactive.function.client.WebClient;
 
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.verifyNoMoreInteractions;
+import static org.mockito.Mockito.when;
+
 /**
  * @author Daniel Lavoie
+ * @author Armin Krezovic
  */
+@MockitoSettings(strictness = Strictness.LENIENT)
 class WebClientTransportClientFactoryTest {
 
+	@Mock
+	private ExchangeFunction exchangeFunction;
+
+	@Captor
+	private ArgumentCaptor<ClientRequest> captor;
+
 	private WebClientTransportClientFactory transportClientFatory;
 
 	@BeforeEach
 	void setup() {
-		transportClientFatory = new WebClientTransportClientFactory(WebClient::builder);
+		ClientResponse mockResponse = mock();
+		when(mockResponse.statusCode()).thenReturn(HttpStatus.OK);
+		when(mockResponse.bodyToMono(Void.class)).thenReturn(Mono.empty());
+		given(exchangeFunction.exchange(captor.capture())).willReturn(Mono.just(mockResponse));
```

Comment:
It's a good approach. On a higher level, you could possibly also use `okhttp3.mockwebserver.MockWebServer` to record and verify the response, but the way you've mocked it is good.